picard/script/__init__.py

```diff
@@ -101,3 +114,236 @@ def enabled_tagger_scripts_texts():
     if not config.setting["enable_tagger_scripts"]:
         return []
     return [(s_name, s_text) for _s_pos, s_name, s_enabled, s_text in config.setting["list_of_scripts"] if s_enabled and s_text]
+
+
+@unique
+class PicardScriptType(Enum):
+    """Picard Script object types
+    """
+    BASE = 0
+    TAGGER = 1
+    FILENAMING = 2
+
+
+class PicardScript():
+    """Base class for Picard script objects.
+    """
+    # Base class developed to support future tagging script class as possible replacement for currently used tuples in config.setting["list_of_scripts"].
+
+    TYPE = PicardScriptType.BASE
+    JSON_OUTPUT = ['title', 'script']
+
+    # Don't automatically trigger changing the `script_last_updated` property when updating these properties.
+    _last_updated_ignore_list = ['last_updated', 'readonly', 'deletable', 'id']
```

Comment:
You can use a set here (but if it is meant to be modified at runtime, which isn't it seems right now):

```suggestion
    _last_updated_ignore_list = {'last_updated', 'readonly', 'deletable', 'id'}
```